export/metric_to_pb.go

```diff
@@ -0,0 +1,162 @@
+	v1 "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	r1 "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"go.opencensus.io/metric/metricdata"
+)
+
+func metricsToServiceRequest(ms []*metricdata.Metric) *a1.ExportMetricsServiceRequest {
+	metrics := []*v1.Metric{}
+
+	for _, m := range ms {
+		toAppend := metricToProto(m)
+		metrics = append(metrics, toAppend)
+	}
+
+	return &a1.ExportMetricsServiceRequest{
+		Metrics: metrics,
+	}
+}
+
+func metricToProto(m *metricdata.Metric) *v1.Metric {
+	return &v1.Metric{
+		MetricDescriptor: metricToDescriptor(m),
+		Timeseries:       metricToTimeSeries(m),
+		Resource:         metricToResource(m),
+	}
+}
+
+func metricToLabelKeys(m *metricdata.Metric) []*v1.LabelKey {
+	labelKeys := []*v1.LabelKey{}
+
+	for _, lk := range m.Descriptor.LabelKeys {
+		toAppend := &v1.LabelKey{
+			Key:         lk.Key,
+			Description: lk.Description,
+		}
+
+		labelKeys = append(labelKeys, toAppend)
+	}
+	return labelKeys
+}
+
+func metricToDescriptor(m *metricdata.Metric) *v1.MetricDescriptor {
+	return &v1.MetricDescriptor{
+		Name:        m.Descriptor.Name,
+		Description: m.Descriptor.Description,
+		Unit:        string(m.Descriptor.Unit),
+		Type:        v1.MetricDescriptor_Type(m.Descriptor.Type + 1),
+		LabelKeys:   metricToLabelKeys(m),
+	}
+}
+
+func metricToResource(m *metricdata.Metric) *r1.Resource {
+	if m.Resource != nil {
+		return &r1.Resource{
+			Type:   m.Resource.Type,
+			Labels: m.Resource.Labels,
+		}
+	}
+
+	return nil
+
+}
+
+func metricToTimeSeries(m *metricdata.Metric) []*v1.TimeSeries {
+	timeSeries := []*v1.TimeSeries{}
+
+	for _, ts := range m.TimeSeries {
+		timestamp, _ := ptypes.TimestampProto(ts.StartTime)
+
+		toAppend := &v1.TimeSeries{
+			StartTimestamp: timestamp,
+			LabelValues:    metricToLabelValues(ts),
+			Points:         metricToPoints(ts),
+		}
+
+		timeSeries = append(timeSeries, toAppend)
+	}
+
+	return timeSeries
+}
+
+func metricToLabelValues(t *metricdata.TimeSeries) []*v1.LabelValue {
+	labelValues := []*v1.LabelValue{}
+
+	for _, lv := range t.LabelValues {
+		toAppend := &v1.LabelValue{
+			Value:    lv.Value,
+			HasValue: lv.Present,
+		}
+
+		labelValues = append(labelValues, toAppend)
+	}
+
+	return labelValues
+}
+
+func metricToPoints(t *metricdata.TimeSeries) []*v1.Point {
+	points := []*v1.Point{}
+
+	for _, p := range t.Points {
+		timestamp, _ := ptypes.TimestampProto(p.Time)
+		toAppend := &v1.Point{
+			Timestamp: timestamp,
+		}
+
+		switch v := p.Value.(type) {
+		case int64:
+			toAppend.Value = &v1.Point_Int64Value{
+				Int64Value: v,
+			}
+		case float64:
+			toAppend.Value = &v1.Point_DoubleValue{
+				DoubleValue: v,
+			}
+		case *metricdata.Summary:
+			toAppend.Value = &v1.Point_SummaryValue{
+				SummaryValue: pointToSummaryValue(v),
+			}
+		// TODO Distribution
```

Comment:
still todo? should we capture a jira?

Reply:
https://confluentinc.atlassian.net/browse/METRICS-2062